src/NServiceBus.Autofac/AutofacObjectBuilder.cs

```diff
@@ -8,34 +8,57 @@ namespace NServiceBus.ObjectBuilder.Autofac
     using global::Autofac.Builder;
     using global::Autofac.Core;
 
+    class ParentBuilder : ContainerBuilder
+    {
+        public ILifetimeScope parentScope;
+
+        public ParentBuilder(ILifetimeScope parentScope)
+        {
+            this.parentScope = parentScope;
+        }
+    }
+
     class AutofacObjectBuilder : Common.IContainer
     {
+        ContainerBuilder builder;
         ILifetimeScope container;
+        HashSet<Type> registeredTypes;
         bool owned;
         bool isChild;
 
-        AutofacObjectBuilder(ILifetimeScope container, bool owned, bool isChild)
+        AutofacObjectBuilder(ContainerBuilder builder, bool owned, bool isChild)
         {
             this.owned = owned;
             this.isChild = isChild;
-            this.container = container;
+            this.builder = builder;
+
+            registeredTypes = new HashSet<Type>();
```

Comment:
The `ContainerBuilder` doesn't actually have a list of the registrations, and they have an issue open since time immemorial to add some kind of functionality https://github.com/autofac/Autofac/issues/469